azurerm/internal/services/databoxedge/databox_edge_device_resource.go

```diff
@@ -0,0 +1,307 @@
+package databoxedge
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/databoxedge/mgmt/2019-08-01/databoxedge"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/location"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/databoxedge/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/databoxedge/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceDataboxEdgeDevice() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDataboxEdgeDeviceCreate,
+		Read:   resourceDataboxEdgeDeviceRead,
+		Update: resourceDataboxEdgeDeviceUpdate,
+		Delete: resourceDataboxEdgeDeviceDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.DataboxEdgeDeviceID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DataboxEdgeName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"sku_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DataboxEdgeDeviceSkuName,
+			},
+
+			"configured_role_types": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"culture": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"device_hcs_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"device_local_capacity": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"device_model": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"device_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"device_software_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"device_type": {
```

Comment:
given there are so man device properties should we put these into a block?

Reply:
Fixed.